javascript/src/unstable.ts

```diff
@@ -237,13 +238,52 @@ export function splice<T>(
     throw new RangeError("invalid object for splice")
   }
   const value = `${objectId}/${prop}`
+  if (typeof index == "string") {
```

Comment:
This feels like it could produce extremely confusing error messages if you were to accidentally pass a string as the `index` but were not intending to pass a cursor. Could we instead use a wrapper class for `Cursor` so we can match using `instanceof` and so that users can't accidentally pass in strings?

Reply:
We could wrap it in a class.  It would make handling it harder - you can't just save it to a database or to disk - we'd need a `toString()` and ` new Cursor(string)` and some documentation.    Maybe instead I ad a regex to make sure the string is in `ctr@actor` form on the branch?